lib/api_repo.js

```diff
@@ -699,28 +716,37 @@ function newIsDirFunc(dirName) {
 }
 
 /**
- * _verifyRepo confirms that api repo source is available.
- *
- * if repoDir is set, it confirms that the directory exists
- *
- * if repoDir is not set, but zipUri is, it downloads the api zip to tmp dir and
- * sets that to repoDir
- *
- * if isGoogleApi is `true`, it confirms that repoDir has 'google' subdirectory
- * @param {function(?Error)} done The callback.
+ * Helper function for _checkRepo. Checks that the repoDirs target is available.
+ * @param {function(?Error)} done A callback to be called when the function
+ *   finishes.
  */
-ApiRepo.prototype._checkRepo = function _checkRepo(done) {
-  var that = this;
-  if (this.repoDir) {
-    var checkDir = newIsDirFunc(this.repoDir);
-    if (this.isGoogleApi) {
-      var checkGoogleDir = newIsDirFunc(path.join(that.repoDir, 'google'));
-      async.waterfall([checkDir, checkGoogleDir], done);
+ApiRepo.prototype._checkRepoDirs = function _checkRepoDirs(done) {
+  var checkDirs = this.repoDirs.map(function(repoDir) {
+    return newIsDirFunc(repoDir);
+  });
+  var checkGoogleDirs = [];
+  if (this.isGoogleApi) {
+    checkGoogleDirs = this.repoDirs.map(function(repoDir) {
+      return newIsDirFunc(path.join(repoDir, 'google'));
+    });
+  }
+  async.waterfall(checkDirs.concat(checkGoogleDirs), function(err) {
+    if (err) {
+      done(err);
```

Comment:
It looks like `async.waterfall(..., done);` is probably just fine (or `function(err) { done(err || null); }`)